common/utils/src/main/resources/error/error-classes.json

```diff
@@ -133,6 +133,48 @@
       "Could not load Protobuf class with name <protobufClassName>. <explanation>."
     ]
   },
+  "CANNOT_LOAD_STATE_STORE" : {
+    "message" : [
+      "An error occurred during loading state."
+    ],
+    "subClass" : {
+      "CANNOT_READ_CHECKPOINT" : {
+        "message" : [
+          "Cannot read RocksDB checkpoint metadata of version <versionLine>"
+        ]
+      },
+      "CANNOT_READ_DELTA_FILE" : {
+        "message" : [
+          "Error reading delta file <fileToRead> of <clazz>: <message>"
+        ]
+      },
+      "CANNOT_READ_SNAPSHOT_FILE" : {
+        "message" : [
+          "Error reading snapshot file <fileToRead> of <clazz>: <message>"
+        ]
+      },
+      "CANNOT_READ_STREAMING_STATE_FILE" : {
+        "message" : [
+          "Error reading streaming state file of <fileToRead> does not exist. If the stream job is restarted with a new or updated state operation, please create a new checkpoint location or clear the existing checkpoint location."
+        ]
+      },
+      "UNEXPECTED_FILE_SIZE" : {
+        "message" : [
+          "Copied <dfsFile> to <localFile>, expected <expectedSize> bytes, found <localFileSize> bytes."
+        ]
+      },
+      "UNRELEASED_THREAD_ERROR": {
+        "message" : [
+          "<loggingId>: RocksDB instance could not be acquired by <newAcquiredThreadInfo> as it was not released by <acquiredThreadInfo> after <timeWaitedMs> ms.",
+          "Thread holding the lock has trace: <stackTraceOutput>"
+        ]
+      },
+      "WRAPPER" : {
+        "message" : [""]
```

Comment:
For the WRAPPER error subclass... @HeartSaVioR suggested we add a high-level wrapping error to wrap the the explicitly thrown and non-explicitly thrown errors. So I added a catch block in getStore and getReadStore where the WRAPPER error is thrown if anything is caught. I'm not sure if "WRAPPER" is the best way to name it though

Reply:
How about OTHER or UNKNOWN

Reply:
I think OTHER or UNKNOWN could be confusing since the error cause may be known. WRAPPER is just a high-level error thrown that encompasses any error thrown underneath it.

For example, throwing cannotReadDeltaFile or fileNotFoundException with getStore/getReadStore as the entry point will then throw the WRAPPER error. We can just throw cannotReadDeltaFile or fileNotFoundException and leave it as that, but wrapping it with the WRAPPER error can give the user more context that the error comes from loading state.

Reply:
How about "uncategorized"? Technically it is categorized as "CANNOT_LOAD_STATE_STORE", but not yet sub-categorized.

Moreover, (I'm not an expert of error class framework, so sorry for the ignorance), is it possible to just use the superclass `CANNOT_LOAD_STATE_STORE`? If then I don't think we need to spot for subclass to handle uncategorized things.

Reply:
UNCATEGORIZED sounds good. Unfortunately, directly using the CANNOT_LOAD_STATE_STORE superclass results in an INTERNAL_ERROR